### PR TITLE
Figure

### PR DESCRIPTION
**Pull request by ataua**
It is done.

Usage:
`<Figure src='srcString' alt='altString' x={Number} y={Number} />`
`x` and `y` are optional and correspond to the width `and` `height` of the inner image;
`alt` will be the `alt` for the image and the text content for the `figcaption`.

### CODE REVIEW

**Comment by ataua** [comment]
## What to do with conflicts:



- my VSCode was buggy so I uninstalled and reinstalled it and updated and changed extensions so some stuff in the project config used my own configuration, bu t you can ignore (not accept changes):
.vscode/settings.json
tsconfig.json
yarn.lock

- Also cancel changes on:
src/components/grid/styles.tsx
(not my component, messed with it to test typescript before adopting on my own stuff)

- Just look conflicts on:
example/src/App.tsx - I think I changed App's folder from src/ to src/__test__.
src/index.test.tsx - I think that's my component's test. If so, leave it as is.
src/index.tsx - I gess it is my component's place. If so, accept as is.

File: example/src/App.jsx [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+      />
+      <Figure
+        src='https://440245-1378968-raikfcquaxqncofqfm.stackpathdns.com/wp-content/uploads/2019/01/fatos-curiosos-sobre-o-homer-simpson.jpg'
+        alt='Homer Simpson'
+      />
+    </>
+  )
+}
+
+export default App
```
Comment (brunocamposal) [inline example/src/App.jsx:29]: Colocar o teste do componente no arquivo App.tsx em vez de App.jsx

File: example/src/__tests__/figure.test.jsx [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+import { Figure } from 'capstone-lib/'
+
+describe('Figure: ', () => {
+  it('renders the component', () => {
+    const tree = renderer
+      .create(<Figure alt='coisa' url='http://teste.coisa.com' y={140} />)
+      .toJSON()
+    expect(tree).toMatchSnapshot()
+  })
+})
```
Comment (brunocamposal) [inline example/src/__tests__/figure.test.jsx:12]: Fazer o teste na pasta src/components/figure/__tests__

File: src/components/figure/styles.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+  display: inline-box;
+  white-space: nowrap;
+  margin: 1px;
+  width: x;
+  height: y;
+`
+export const StyledFigure = styled.figure`
+  box-sizing: content-box;
+  flex: 1;
+  border: 1px solid #666;
```
Comment (brunocamposal) [inline src/components/figure/styles.js:14]: Alterar extensão do arquivo para styles.txs

**Review by brunocamposal — COMMENTED** [review]
Por gentileza, fazer as alterações dos comentários que adicionei

**Pull request closed without merging**